uisdk/src/main/java/com/karhoo/uisdk/screen/booking/checkout/payment/BookingPaymentHandler.kt

```diff
@@ -121,6 +123,16 @@ class BookingPaymentHandler @JvmOverloads constructor(
     }
 
     override fun retrieveLoyaltyStatus() {
-        paymentActions?.retrieveLoyaltyStatus()
+        var loyaltyEnabled = true
+        val featureFlag = FeatureFlagsProvider.getFeatureFlags()
+        featureFlag?.forEach {
+            if(BuildConfig.VERSION_NAME >= it.version){
+                if (it.flags[FeatureFlagsProvider.LOYALTY_ENABLED] == false)
+                    loyaltyEnabled = false
+            }
+        }
+
+        if(loyaltyEnabled && userStore.paymentProvider?.loyalty?.id.isNullOrEmpty())
```

Comment:
if(loyaltyEnabled && **NOT** userStore.paymentProvider?.loyalty?.id.isNullOrEmpty()). If the id is null or empty, then the feature is disabled. The status should be retrieved in the other case